azure-linux.dockerfile

```diff
@@ -8,7 +8,12 @@ FROM $IMAGE
 
 # ca-certificates: Azure Linux base image does not contain Mozilla CA certificates, install it to prevent CERTIFICATE_VERIFY_FAILED errors, see https://github.com/Azure/azure-cli/issues/26026
 # jq: It's widely used for parsing JSON output in Azure CLI and has a small size. See https://github.com/Azure/azure-cli/issues/29830
-RUN --mount=type=bind,target=/azure-cli.rpm,source=./docker-temp/azure-cli.rpm tdnf install ca-certificates jq /azure-cli.rpm -y && tdnf clean all && rm -rf /var/cache/tdnf
+RUN --mount=type=bind,target=/azure-cli.rpm,source=./docker-temp/azure-cli.rpm \
+    tdnf install ca-certificates jq /azure-cli.rpm -y && \
+    # Run az version to generate the commandIndex.json and speed up the following az commands
+    # Keep versionCheck.json to prevent az from checking the latest version on GitHub
```

Comment:
It always tries to get latest version from GitHub. I believe it's not necessary.

This reminds me of https://github.com/Azure/azure-cli/issues/19183.

Ref: https://github.com/Azure/azure-cli/blob/a29d2e78ac30942b98427627e4f2c4a598bc1eeb/src/azure-cli-core/azure/cli/core/__init__.py#L85